packages/@aws-cdk/aws-s3/README.md

```diff
@@ -249,6 +249,23 @@ const bucket = s3.Bucket.fromBucketAttributes(this, 'ImportedBucket', {
 bucket.addEventNotification(s3.EventType.OBJECT_CREATED, new s3n.SnsDestination(topic));
 ```
 
+When you add an event notification to a bucket, a custom resource is created to 
+manage the notifications. By default, a new role is created for the Lambda
+function that implements this feature. If you want to use your own role instead,
+you should provide it in the `Bucket` constructor:
+
+```ts
+declare const importedRole: iam.IRole;
+const bucket = new s3.Bucket(this, 'MyBucket', {
+  notificationsHandlerRole: importedRole,
+});
+```
+
+> NOTE: If you provide your own handler role, make sure that it has at least
```

Comment:
I think this should be revised a little bit. Generally in CDK passing an `IRole` doesn't mean that no policies can be added to, since an IAM policy is a separate resource, we do often add such policies even to imported roles. 

The way a user would indicate that the `IRole` **cannot** be modified at all, is by passing the `mutable: false` property when importing the role.

https://github.com/aws/aws-cdk/blob/534babde886646b6cd01f8799e3b4ed6db221263/packages/%40aws-cdk/aws-iam/lib/role.ts#L147

Doing so will void any policy operation on this role, and in such case, the role must already be configured with the right permissions. 

So, I would revise this to `if you provide your own handler role, and configure it to be immutable using the `mutable: false` property...`

Would also emphasize that simply passing an `IRole` does not guarantee immutability.